### PR TITLE
2nd batch of tests for the matching engine

### PR DESCRIPTION
**Pull request by gauthamk97**

### CODE REVIEW

File: matchingengine/OrderBook_test.go [hunk 1]
```diff
@@ -53,3 +53,197 @@ func TestLoadOldAsk(t *testing.T) {
...
+		Id:            2,
+		UserId:        10,
+		StockId:       stockID,
+		Type:          2,
+		StockQuantity: 10,
+		Price:         20,
+		Total:         200,
+		CreatedAt:     "random2",
+	}
+
```
Comment (coderick14) [inline matchingengine/OrderBook_test.go]: Can we have a helper function `makeTransaction` just like `makeAsk`/`makeBid`?

File: matchingengine/OrderBook_test.go [hunk 2]
```diff
@@ -53,3 +53,197 @@ func TestLoadOldAsk(t *testing.T) {
...
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	stopLossAsk := makeAsk(1, stockID, models.StopLoss, stockQuantity, stockPrice, "")
+
+	mockAskStoplossQueue.EXPECT().Push(stopLossAsk).Times(1)
+
```
Comment (coderick14) [inline matchingengine/OrderBook_test.go]: Does removing the check for `AddOrder` ensure that the test fails if it is called? Or do we need to add `Times(0)` in order to ensure it?

File: matchingengine/OrderBook_test.go (hunk 1 above)
Comment (gauthamk97) [reply matchingengine/OrderBook_test.go]: Done for Transaction. For now, let's create order books the way we're doing now.

File: matchingengine/OrderBook_test.go [hunk 3]
```diff
@@ -53,3 +67,178 @@ func TestLoadOldAsk(t *testing.T) {
...
+	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
```
Comment (PaiAkshay998) [inline matchingengine/OrderBook_test.go]: ```stockId := 1```

Comment (gauthamk97) [reply matchingengine/OrderBook_test.go]: that will make stockId as an `int`. I used this to cast it to `uint32` which is what it needs to be.

Comment (PaiAkshay998) [reply matchingengine/OrderBook_test.go]: Won't ```stockId := uint32(1)``` work? I mean I really don't care about using ```var``` or ```:=``` notation but just putting it there :stuck_out_tongue:

**Pull request merged**